Assets/_Project/Scripts/Map/LocationProviderMapUpdater.cs

```diff
@@ -0,0 +1,97 @@
+using System.Collections;
+using System.Collections.Generic;
+using Mapbox.Unity.Location;
+using Mapbox.Unity.Map;
+using Mapbox.Utils;
+using PolSl.UrbanHealthPath.Navigation;
+using UnityEngine;
+using ILocationProvider = PolSl.UrbanHealthPath.Navigation.ILocationProvider;
+
+namespace PolSl.UrbanHealthPath
+{
+    public class LocationProviderMapUpdater : MonoBehaviour
```

Comment:
Why MonoBehaviour?

Reply:
Doesn't it need to be a MB to be attacked to the map as it updates the map constantly?

Reply:
From what I see, you need it only for coroutines (they have to be attached to MBs). There could be a more elegant solution (maybe a CoroutineManager game object and passing it to a constructor) but I would leave it as it is for now.